reference/tools.rst

```diff
@@ -810,19 +810,31 @@ tools.cross_building()
 
 .. code-block:: python
 
-    def cross_building(settings, self_os=None, self_arch=None, skip_x64_x86=False)
+    def cross_building(conanfile, self_os=None, self_arch=None, skip_x64_x86=False)
+
+
+Evaluates operating system and architecture from the ``host`` machine and the ``build`` machine
+to return a boolean ``True`` if it is a cross building scenario. Settings from ``host`` machine are
+taken from the ``conanfile.settings``, while setting from the ``build`` context can provide from
+different sources:
 
-Reading the settings and the current host machine it returns ``True`` if we are cross building a Conan package:
+* if ``conanfile.settings_build`` is available (Conan was called with a ``--profile:build``) it will
+  use settings in that profile (read more about :ref:`build_requires_context`).
+* otherwise, the values for the ``build`` context will come from (in this order of precedence):
+  ``self_os`` and ``self_arch`` if they are given to the function, the values for ``os_build``
+  and ``arch_build`` from ``conanfile.settings`` or auto-detected. 
+
+This tool can be used to run special actions depending on its return value:
 
 .. code-block:: python
 
     from conans import tools
 
-    if tools.cross_building(self.settings):
+    if tools.cross_building(self):
         # Some special action
 
 Parameters:
-    - **settings** (Required): Conanfile settings. Use ``self.settings``.
+    - **conanfile** (Required): Conanfile object. Use ``self`` in a ``conanfile.py``.
```

Comment:
Is this backward compatible with a `settings` argument as well?

Reply:
No, we don't want them to be passing settings anymore. The argument in latest Conan is a conanfile. Yes, it might not break, but that is a different story.